From 99f7803b5e18fb6b9b51a495f911d70413d0fccc Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Fri, 1 Sep 2017 16:56:34 +0800
Subject: [PATCH] PM / devfreq: rockchip_dmc: Fix min_freq

Auto_min_rate is used to instead of normal_rate, if other sys_status is
exist, use status_rate as min_freq, otherwise use auto_min_rate.

Change-Id: If2df35396ec9e6038f18b173a78ff1a248ab87eb
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 drivers/devfreq/rockchip_dmc.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/devfreq/rockchip_dmc.c b/drivers/devfreq/rockchip_dmc.c
index 5dc02d41715b..b6d94b5e8182 100644
--- a/drivers/devfreq/rockchip_dmc.c
+++ b/drivers/devfreq/rockchip_dmc.c
@@ -1046,7 +1046,10 @@ static void rockchip_dmcfreq_update_target(struct rockchip_dmcfreq *dmcfreq)
 	}
 
 	if (dmcfreq->auto_freq_en) {
-		min_freq = max(dmcfreq->status_rate, dmcfreq->auto_min_rate);
+		if (dmcfreq->status_rate)
+			min_freq = dmcfreq->status_rate;
+		else if (dmcfreq->auto_min_rate)
+			min_freq = dmcfreq->auto_min_rate;
 		if (!min_freq)
 			min_freq = dmcfreq->min;
 		max_freq = dmcfreq->max;
-- 
2.35.3

